File: vigenere.py
import sys
if int(len(sys.argv)) != 2 or sys.argv[1].isalpha() == False:
    sys.exit("Usage: python vigenere.py k")
key = sys.argv[1].upper()
k = len(key)
index_k = 0
text = input("plaintext: ")
print("ciphertext: ", end="")
for i in range (len(text)):
    if text[i].isupper():
        print(chr(65 + (ord(text[i]) - 65 + ord(key[index_k % k]) - 65) % 26), end="")
        index_k += 1
    elif text[i].islower():
        print(chr(97 + (ord(text[i]) - 97 + ord(key[index_k % k]) - 65) % 26), end="")
        index_k += 1
    else:
        print(text[i], end="")
print()

File: vigenere.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    int invalid;
    string key;
    string text;
        // checking the number of arguments
    if (argc > 2 || argc ==1)
    {
        printf("Usage: ./vigenere keyword \n");
        return 1;
    }
        // checking the line item's alphabet
    for (int i = 0, n = strlen(argv[1]); i < n; i++)
    {
        if (isalpha(argv[1][i]))
        {
            invalid = 0;
        }
        else
        {
            invalid = 1;
        }
    }
    if (invalid == 1)
    {
        printf ("Usage: ./vigenere keyword\n");
        return 1;
    }
        // get text for encryption
    else
    {
        text = get_string("plaintext: ");
    }

        // lengh string key
    key = argv[1];
    for (int i = 0, n = strlen(argv[1]); i < n; i++)
    {
        key[i] = toupper(argv[1][i]);
    }
    int l_key = strlen(key);

        // encryption the text
    printf ("ciphertext: ");
    int index_k = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            if (isupper(text[i]))
            {
                int upper = 'A' + (((text[i] - 65) + (key[index_k%l_key] - 65))%26);
                printf ("%c", (char)upper);
                index_k++;
            }
            if (islower(text[i]))
            {
                int lower = 'a' + (((text[i] - 97) + (key[index_k%l_key] - 65))%26);
                printf ("%c", (char)lower);
                index_k++;
            }
        }
        else
        {
            printf ("%c", text[i]);
        }
    }
    printf ("\n");

    return 0;
}
